exploration/text-vs-code.md

```diff
@@ -136,16 +209,93 @@ The current syntax includes some plain-ascii keywords:
 `input`, `local`, `match`, and `when`.
 
 The current syntax and active proposals include some sigil + name combinations,
-such as `:number`, `$var`, `|literal|`, `+bold`, and `@attr`.
+such as `:number`, `$var`, `|literal|`, `+bold`, `-bold`, and posibly `@attr`.
 
 The current syntax supports unquoted literal values as operands.
 
-Messages themselves are "simple strings" and must be considered to be a single
-line of text. In many containing formats, newlines will be represented as the local
-equivalent of `\n`.
+Messages themselves are "simple strings" and must be considered to be WYSIWYG.
+The WYSIWYG nature of representing a message pattern is independent of whether the message is a single line or contains multiple lines.
+
+There is no restriction that a message must only contain a single line (that is, not contain any newline characters),
+nor are there constraints about how newlines must be represented. As our [`spec/syntax.md`](../spec/syntax.md) states:
+
+> Any Unicode code point is allowed, except for surrogate code points U+D800 through U+DFFF inclusive.
+
+> Whitespace in _text_, including tabs, spaces, and newlines is significant and MUST be preserved during formatting.
+
+> ... Instead, we tolerate direct use of nearly all
+characters (including line breaks, control characters, etc.) and rely upon escaping
+in those outer formats to aid human comprehension (e.g., depending upon container
+format, a U+000A LINE FEED might be represented as `\n`, `\012`, `\x0A`, `\u000A`,
+`\U0000000A`, `&#xA;`, `&NewLine;`, `%0A`, `<LF>`, or something else entirely).
+
+## Simple Messages
+
+In the Subcommittee meetings following Github discussions on Issues #493 and #499,
+the general consensus that formed for simple messages
+is that we allow them to be unquoted.
+
+("Simple messages" refers to messages consisting solely of a pattern, and thus are not complex messages.)
+
+Because the simple message pattern consists of the entire message,
+the pattern includes any leading or trailing whitespace.
+
+Given simple messages already being decided at a high level,
+the design decisions below for the proposed and alternative designs pertain specifically to complex messages.
 
 ## Proposed Design
 
+### Start in text, encapsulate message, always quote patterns
+
+Description:
+
+Since simple messages are unquoted (starting in text mode),
+complex messages must also start in text mode.
+
+Within a complex message, patterns are always quoted with `{{...}}` or other choice of delimiter.
+
+The entire complex message is also wrapped with `{{...}}` or other choice of delimiter.
+This allows interior "code mode" of message to have flexible whitespace in between tokens
```

Comment:
What do you think should happen to whitespace outside the entire code block? Should we specify it here as well?

Reply:
We think they should be preserved "as is". If it is not specified, we should.

* consistency with complex (no trimming)
* many of the templating systems don't trim by default, some have it opt-in
* clear what you get
   If you see "   Foo" you know you will see spaces. If we trim ascii spaces, the leading spaces might have nbsp, em-space, en-space, ideographic spaces, etc. So if you read or even edit the string, you can't tell what the result is without a way to inspect those spaces that look all the same.
* as a translator, if I want "  Bill Gates ..." (honorific spaces in Chinese) I don't know if I need to wrap / escape or not.
   Because as a translator I don't know what API is used on that string. Maybe it is MF2, or maybe not.

Reply:
If there is anything "outside" the complex message-wide delimiters, then:

1. It's whitespace that we trim
   a. Or produces an error (worth discussion)
3. or the message is actually a simple message that produces a lot of errors

```
This {{ match {$var} when * {{{$var}}}}} is an interesting message.
```

This evaluates, I think, as:

>This {&#xfffd;} is an interesting message.

Or possibly (with `$var`==123) as:

> This {&#xfffd;}{&#xfffd;} match 123 when * 123{&#xfffd;}{&#xfffd;} is an interesting message.

(both emit a syntax error)